docs/scripts_md/File.md

```diff
@@ -0,0 +1,178 @@
+# NAME
+
+NeuroDB::File -- Provides an interface to the MRI file management subsystem of
+LORIS
+
+# SYNOPSIS
+
+    use NeuroDB::File;
+    use NeuroDB::DBI;
+
+    my $dbh = NeuroDB::DBI::connect_to_db();
+
+    my $file = NeuroDB::File->new(\$dbh);
+
+    my $fileID = $file->findFile('/path/to/some/file');
+    $file->loadFile($fileID);
+
+    my $acquisition_date = $file->getParameter('acquisition_date');
+    my $parameters_hashref = $file->getParameters();
+
+    my $coordinate_space = $file->getFileDatum('CoordinateSpace');
+    my $filedata_hashref = $file->getFileData();
+
+
+    # less common to use methods, available mainly for register_db...
+    my $dbh_copy = $file->getDatabaseHandleRef();
+
+    $file->loadFileFromDisk('/path/to/some/file');
+    $file->setFileData('CoordinateSpace', 'nonlinear');
+    $file->setParameter('patient_name', 'Larry Wall');
+
+    my $parameterTypeID = $file->getParameterTypeID('patient_name');
+
+# DESCRIPTION
+
+This class defines a BIC MRI (or related) file (minc, bicobj, xfm,
+etc) as represented within the LORIS database system.
+
+**Note:** if a developer does something naughty (such as leaving out
+the database handle ref when instantiating a new object or so on) the
+class will croak.
+
+## Methods
+
+### new(\\$dbh) >> (constructor)
+
+Create a new instance of this class. The parameter `\$dbh` is a
+reference to a DBI database handle, used to set the object's database
+handle, so that all the DB-driven methods will work.
+
+INPUT: DBI database handle.
+
+RETURNS: new instance of this class.
+
+### loadFile($fileID)
+
+Load the object with all the data pertaining to a file as defined by
+parameter `$fileID`.
+
+INPUT: ID of the file to load.
+
+RETURNS: 0 if no file was found, 1 otherwise.
+
+### findFile($filename)
+
+Finds the FileID pertaining to a file as defined by parameter `$filename`,
+which is a full /path/to/file.
+
+INPUT: full path to the file to look for an ID in the database.
+
+RETURNS: (int) FileID or undef if no file was found.
+
+### getFileData()
+
+Gets the set of file data (data from the `files` table in the database).
+
+INPUT: none
```

Comment:
should we say this or not have INPUT at all? to be consistent with the ones below that do not have an input?

Reply:
good catch!! I'll remove it